dragonfly_doe2/doe/mats_layers_constr.py

```diff
@@ -42,43 +44,91 @@ def lbt_to_ip(self, _val, _to, _from):
+                '   RESISTANCE     = {r_val}\n   ..\n'.format(
+                    r_val=self.lbt_to_ip([self.hbmat.r_value], 'h-ft2-F/Btu', 'm2-K/W'))
 
     def __repr__(self):
         return self.to_inp()
 
 
-class Construction(object):
+@dataclass()
+class MatsLayersConstructions(object):
     """Construction object. Contains, materials and layers for *.inp file.
     Intent: Pass list of constrs in dfm to this class.
     Should return enum of mats, 'layers' (stupid f'n eQuest obj imo..), Constr objs
-    via __repr__
-    TODO: make it happen! easy day! tomorrow.
+    via __repr__.
+
+        Returns:
+          $  Materials / Layers / Constructions *.inp block
     """
+    inp_constructions: list = None
+    inp_layers: list = None
+    inp_materials: list = None
 
-    def __init__(self, hb_constructions: list):
-        self._hb_constructions = hb_constructions
+    @classmethod
+    def from_hb_constructions(cls, hb_constructions: list):
+        """Create input data for eQuest $ Materials / Layers / Constructions *.inp block"""
+        roughdict = {'VeryRough': 1, 'Rough': 2, 'MediumRough': 3,
+                     'MediumSmooth': 4, 'Smooth': 5, 'VerySmooth': 6}
+        cls_ = cls()
+        inp_cons = []
+        inp_layrs = []
+        inp_mats = []
+        for i, constr in enumerate(hb_constructions):
+            if type(constr) == OpConstr:
+                mats = [mat for mat in constr.materials]
 
-    @property
-    def hb_constructions(self):
-        return self._hb_constructions
+                for mat in mats:
+                    inp_mats.append(Material(mat))
+
+                matstr = ''.join(f'"{m.display_name}", ' for m in mats)
+                inp_layrs.append(
+                    f'"{constr.display_name}_{i} Layers" = LAYERS\n'
+                    f'   MATERIAL                 = ( {matstr} )\n   ..\n')
+
+                inp_cons.append(
+                    f'"{constr.display_name}_{i}" = CONSTRUCTION\n'
+                    f'   TYPE                 = LAYERS\n'
+                    f'   ABSORPTANCE          = {constr.materials[0].solar_absorptance}\n'
+                    f'   ROUGHNESS            = {roughdict[constr.materials[0].roughness]}\n'
+                    f'   LAYERS               = "{constr.display_name}_{i} Layers"\n   ..\n')
 
-    # ...STATICMETHOD!! noo... nooo... ...maybe. unitl tomoz!
+        cls_.inp_constructions = inp_cons
+        cls_.inp_layers = inp_layrs
+        cls_.inp_materials = inp_mats
+        return(cls_)
+
+    def to_inp(self):
+        # ? if these could be list comps that would be chill,
+        # TODO: Look at replacing with lambda func
+
+        block = fb.matslayers
+        for mat in self.inp_materials:
+            block += mat.to_inp()
+        for lay in self.inp_layers:
+            block += lay
+        for cons in self.inp_constructions:
+            block += cons
+        return(block)
```

Comment:
I don't know why you do this but it making me crazy! 😄 - no `( )` is needed

```suggestion
        return block
```

Reply:
10:4 solid copy!! I started doing this a while ago; and not even consistently, which makes it extra annoying imo 😅.
Will make extra endeavor to ```STOP``` !